pygmt/clib/loading.py

```diff
@@ -37,22 +32,21 @@ def load_libgmt(env=None):
         couldn't access the functions).
 
     """
-    if env is None:
-        env = os.environ
-    libnames = clib_name(os_name=sys.platform)
-    libpath = env.get("GMT_LIBRARY_PATH", "")
+    lib_fullnames = clib_full_names()
     error = True
-    for libname in libnames:
+    for libname in lib_fullnames:
         try:
-            libgmt = ctypes.CDLL(os.path.join(libpath, libname))
+            libgmt = ctypes.CDLL(libname)
             check_libgmt(libgmt)
             error = False
             break
         except OSError as err:
             error = err
     if error:
         raise GMTCLibNotFoundError(
```

Comment:
Test coverage missing for this section (Lines 43,44,46). Will need to edit and add back the deleted `test_load_libgmt_fail`.

Reply:
The `load_libgmt` function won't fail, as I removed the parameter *env* from this function. Perhaps I should add *env* back, so we can pass a "fake" _env_?

Reply:
Maybe remove the `try... except` clause then since there won't be an OSError?

Reply:
Wait, scratch that. If you set a fake `GMT_LIBRARY_PATH` using `os.environ["GMT_LIBRARY_PATH"] = "/not/a/real/path"`, that would be the same as passing in a fake env right?